src/profile-logic/marker-data.js

```diff
@@ -602,14 +608,19 @@ export function filterRawMarkerTableToRange(
   return newMarkerTable;
 }
 
-export function filterMarkersToRange(
-  markers: Marker[],
+export function filterMarkerRefsToRange(
+  referenceTable: Marker[],
+  markerRefs: MarkerRef[],
   rangeStart: number,
   rangeEnd: number
-): Marker[] {
-  return markers.filter(
-    tm => tm.start < rangeEnd && tm.start + tm.dur >= rangeStart
-  );
+): MarkerRef[] {
+  return markerRefs.filter(ref => {
+    const marker = referenceTable[ref];
+    if (!marker) {
+      return false;
+    }
+    return marker.start < rangeEnd && marker.start + marker.dur >= rangeStart;
+  });
```

Comment:
I use this pattern in a lot of places:
```js
return refs.filter(ref => {
    const marker = referenceTable[ref];
    if (!marker) {
      return false;
    }
    return filterFunction(marker);
  });
```

Do you think I should extract it, eg to a `filterMarkerRef(referenceTable, refs, func)`?